### PR TITLE
Compile performance optimizations

### PR DESCRIPTION
**Pull request by Rakoth**
Optimised regular expressions for interpolation and inline tags.
Some performance tests are provided, but I am not sure if I should keep them in repo.
Each test case took more than 200ms on my computer.
Any thoughts?

### CODE REVIEW

**Comment by doomspork** [comment]
Nice work @Rakoth, thanks for taking the initiative on this.  

Good question about the tests, I'm not sure how best to handle that in an Elixir project.  Any thoughts @lpil or @henrik?

File: test/parser_test.exs [hunk 1]
```diff
@@ -122,6 +122,14 @@ defmodule ParserTest do
     assert opts[:content] == ~s(" text \#{content}\n")
   end
 
+  test "interpolation performance on long lines" do
+    slime = ~S(h2 Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do #{"eiusmod"})
+    {time, _} = :timer.tc(fn ->
+      Parser.parse_line(slime)
+    end)
+    assert time < 1000
```
Comment (lpil) [inline test/parser_test.exs]: I don't understand how this is a useful test. It varies depending on the machine, the current workload, etc.

**Comment by lpil** [comment]
We can add benchmarks one can run, but I don't see the value in these tests being kept in the repo.

**Comment by Rakoth** [comment]
Agreed, @lpil

**Comment by doomspork** [comment]
Should we merge this and release `0.12.0`?

**Comment by lpil** [comment]
Sounds good. :)

**Pull request merged**